### PR TITLE
02-testing, unit and integration testing

### PR DESCRIPTION
**Pull request by danetsao**
Added unit testing and integration testing via [training](https://github.com/ualibweb/ua-folio-docs/blob/main/training/backend/02-testing.md).

Did not include testing for BookMapper, but acheived 100% code coverage on all other elements(BookService, BookController, Book).

### CODE REVIEW

File: src/main/java/org/folio/sample/domain/mapper/BookMapper.java [hunk 1]
```diff
@@ -15,4 +15,6 @@ public interface BookMapper {
 
   @Mapping(target = "id", ignore = true)
   Book fromDto(BookForCreationDTO source);
+
+BookDTO bookForCreationDTOToBookDTO(BookForCreationDTO bookForCreationDTO);
```
Comment (ncovercash) [inline src/main/java/org/folio/sample/domain/mapper/BookMapper.java]: Any reason this was added? I don't see that this is used

**Review by ncovercash — APPROVED** [review]
Looks good to me. @nhanaa please review this

File: src/main/java/org/folio/sample/domain/mapper/BookMapper.java (hunk 1 above)
Comment (danetsao) [reply src/main/java/org/folio/sample/domain/mapper/BookMapper.java]: Yea no reason, I added by mistake. I removed it now.

**Review by nhanaa — APPROVED** [review]
Looks great!

**Pull request merged**